extensions/panache/mongodb-panache/deployment/src/main/java/io/quarkus/mongodb/panache/deployment/PanacheResourceProcessor.java

```diff
@@ -43,19 +41,22 @@ FeatureBuildItem featureBuildItem() {
     }
 
     @BuildStep
-    AdditionalBeanBuildItem registerJacksonSerDer(Capabilities capabilities) {
-        if (capabilities.isCapabilityPresent("io.quarkus.resteasy.jackson")) {
-            return AdditionalBeanBuildItem.unremovableOf(ObjectMapperProducer.class);
-        }
-        return null;
+    void registerJsonbSerDeser(BuildProducer<JsonbSerializerBuildItem> jsonbSerializers,
+            BuildProducer<JsonbDeserializerBuildItem> jsonbDeserializers) {
+        jsonbSerializers
+                .produce(new JsonbSerializerBuildItem(io.quarkus.mongodb.panache.jsonb.ObjectIdSerializer.class.getName()));
+        jsonbDeserializers
+                .produce(new JsonbDeserializerBuildItem(io.quarkus.mongodb.panache.jsonb.ObjectIdDeserializer.class.getName()));
     }
 
     @BuildStep
-    ResteasyJaxrsProviderBuildItem registerJsonbSerDer(Capabilities capabilities) {
-        if (capabilities.isCapabilityPresent("io.quarkus.resteasy.jsonb")) {
```

Comment:
You remove the check to the capability presence, did you check that an apps without jsonb and jackson still works ?
I remind having classpath issue without this check

Reply:
Yup. This now works because of the build items being in the SPI module (which has no dependency on the actual jackson/jsonb module). So even if Jsob and / Jackson are not present, the SPI build items will just be dropped on the floor. 
This sort of pattern is very flexible and has been used in other parts of Quarkus as well (see the Kubernetes or Undertow extensions) for example.

Reply:
OK, thanks for the explaintation